include/data.h

```diff
@@ -71,6 +71,8 @@ struct TrainerMon
     bool8 isShiny : 1;
     u8 dynamaxLevel : 4;
     bool8 gigantamaxFactor : 1;
+    bool8 shouldDynamax : 1;
```

Comment:
Can the dynamaxLevel not be used in this case?

Reply:
I'm not entirely sure this could be used, as one could set the dynamaxLevel to 0 and still do dynamax for 50% HP increase (see https://bulbapedia.bulbagarden.net/wiki/Dynamax#Dynamax_Level_and_HP).
Setting the dynamax level would clearly mean that the Pokémon is supposed to dynamax, however this not does imply the reverse.